patches/api/0463-add-overloads-to-use-suspicious-stew-entry-to-mushro.patch

```diff
@@ -0,0 +1,49 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Yannick Lamprecht <yannicklamprecht@live.de>
+Date: Mon, 12 Feb 2024 09:17:28 +0100
+Subject: [PATCH] add overloads to use suspicious stew entry to mushroom cow
+
+The existing method with PotionEffect suggests that all attributes are used. In fact, only the PotionEffectType and the duration are used.
+
+diff --git a/src/main/java/org/bukkit/entity/MushroomCow.java b/src/main/java/org/bukkit/entity/MushroomCow.java
+index 2f9aefd38c43755c79b30abddd6643b26880bd0d..1085bd85d52cd3b3e5d1481baa00858f7fd6daf6 100644
+--- a/src/main/java/org/bukkit/entity/MushroomCow.java
++++ b/src/main/java/org/bukkit/entity/MushroomCow.java
+@@ -1,5 +1,6 @@
+ package org.bukkit.entity;
+ 
++import io.papermc.paper.potion.SuspiciousEffectEntry;
+ import java.util.List;
+ import org.bukkit.potion.PotionEffect;
+ import org.bukkit.potion.PotionEffectType;
+@@ -27,6 +28,7 @@ public interface MushroomCow extends Cow, io.papermc.paper.entity.Shearable { //
+      *
+      * @return an immutable list of custom potion effects
+      */
++    @Deprecated
+     @NotNull
+     List<PotionEffect> getEffectsForNextStew();
+ 
+@@ -40,8 +42,13 @@ public interface MushroomCow extends Cow, io.papermc.paper.entity.Shearable { //
+      * @return true if the effects to be applied to the suspicious stew changed
+      * as a result of this call
+      */
++    @Deprecated
+     boolean addEffectToNextStew(@NotNull PotionEffect effect, boolean overwrite);
+ 
++    boolean addEffectToNextStew(@NotNull SuspiciousEffectEntry suspiciousEffectEntry, boolean overwrite);
++
++    boolean hasEffectForNextStew(@NotNull SuspiciousEffectEntry suspiciousEffectEntry);
++
+     /**
+      * Removes a custom potion effect from being applied to the next suspicious
+      * stew received from milking this {@link MushroomCow}.
+@@ -52,6 +59,8 @@ public interface MushroomCow extends Cow, io.papermc.paper.entity.Shearable { //
+      */
+     boolean removeEffectFromNextStew(@NotNull PotionEffectType type);
+ 
++    boolean removeEffectFromNextStew(@NotNull SuspiciousEffectEntry suspiciousEffectEntry);
```

Comment:
I think a similar thing applies here, where we don't necessarily need a remove method that takes an entry.

Reply:
Depends on the requirement. Do we ever want to remove a potion effect type with a specific duration and keep the same potion effect types with different duration?